jwst/assign_wcs/util.py

```diff
@@ -615,25 +615,24 @@ def create_grism_bbox(input_model,
                     # location of the +/- sides of the bounding box in the
                     # grism image
                     lmin, lmax = range_select.pop()
-                    xmin, ymin, _, _, _ = sky_to_grism(obj.sky_bbox_ll.ra.value,
-                                                       obj.sky_bbox_ll.dec.value,
-                                                       lmin, order)
-                    xmax, ymax, _, _, _ = sky_to_grism(obj.sky_bbox_ur.ra.value,
-                                                       obj.sky_bbox_ur.dec.value,
-                                                       lmax, order)
-
-                    # Make sure that we have the correct box corners tagged
-                    # as the min and max extents for the grism, the dispersion
-                    # direction changes with grism and detector
-                    if xmax < xmin:
-                        xmin, xmax = xmax, xmin
-                    if ymax < ymin:
-                        ymin, ymax = ymax, ymin
-
-                    xmin = int(xmin)
-                    xmax = int(xmax)
-                    ymin = int(ymin)
-                    ymax = int(ymax)
+                    ra = np.array([obj.sky_bbox_ll.ra.value, obj.sky_bbox_lr.ra.value,
+                                   obj.sky_bbox_ul.ra.value, obj.sky_bbox_ur.ra.value])
+                    dec = np.array([obj.value for obj in [obj.sky_bbox_ll.dec, obj.sky_bbox_lr.dec,
+                                                          obj.sky_bbox_ul.dec, obj.sky_bbox_ur.dec]])
+                    x1, y1, _, _, _ = sky_to_grism(ra, dec, [lmin] * 4, [order]*4)
+                    x2, y2, _, _, _ = sky_to_grism(ra, dec, [lmax] * 4, [order]*4)
+
+                    xstack = np.hstack([x1, x2])
+                    ystack = np.hstack([y1, y2])
+                    xmin = np.min(xstack)
+                    xmax = np.max(xstack)
+                    ymin = np.min(ystack)
+                    ymax = np.max(ystack)
+
+                    xmin = int(xmin) - input_model.meta.subarray.xstart
+                    xmax = int(xmax) - input_model.meta.subarray.xstart
+                    ymin = int(ymin) - input_model.meta.subarray.ystart
+                    ymax = int(ymax) - input_model.meta.subarray.ystart
```

Comment:
@sosey It seems `xmin`, `ymin`, ..., are in full frame coordinates. Since the bounding box is constructed form them  the origin of the subarray needs to be subtracted. Does this sound right? I haven't see a simulated data set with a subarray to test it.

Reply:
The APT NIRCam WFSS template has the subarray choice hardwired to "FULL" and the NIRISS WFSS template doesn't even have a subarray field (hence default is FULL). So you may not need to worry about this.

Reply:
If I remember correctly, subarrays are only allowed in nircam tsgrism mode. The polynomial transforms only work with the full frame coordinates. The code here is called during extract_2d, and is creating bounding boxes which should be in the full frame coordinates, it just uses the input catalog and the magnitude to limit the objects  that need bounding boxes.  

tsgrism is always supposed to have the source object at the same pixel, and that is hardcoded into the transforms. At least a while ago, the 2d extraction for tsgrism mode didn't call this bounding box code. So I think you're safe to leave the subarray subtraction out.  You might add a note here for the benefit of future code peepers if you agree :)

Reply:
@sosey Another quick question - what's the reason for having these as integer values?

Reply:
After looking at the code again, the bounding boxes of the slits are created from the shape of the extracted array. The `xmin`, ..., values are used to define the extraction indices of the arrays, so that's why they need to be integers. So ignore my question.

Reply:
I think it was just to give an integer pixel value for the bounding box corners. If that doesn't matter you can exclude the int. I just noticed that later on there is a round() called as well. I think both are definitely not needed if you decide to keep the integer value in the order bounding.